src/components/Menu/FaucetModal.tsx

```diff
@@ -0,0 +1,182 @@
+import { getFullDisplayBalance } from 'utils/formatBalance'
+import { BigNumber } from 'ethers'
+import { useAllTokens } from 'hooks/Tokens'
+import { filterTokens } from 'components/SearchModal/filtering'
+import Logo from 'components/Logo'
+import { logo } from 'components/CurrencyLogo'
+import useMixpanel, { MIXPANEL_TYPE } from 'hooks/useMixpanel'
+const AddressWrapper = styled.div`
+  background: ${({ theme }) => theme.buttonBlack};
+  border-radius: 8px;
+  padding: 12px;
+  overflow: hidden;
+  p {
+    margin: 12px 0 0 0;
+    font-size: 24px;
+    line-height: 28px;
+    font-weight: 500;
+    color: ${({ theme }) => theme.disableText};
+  }
+`
+function FaucetModal() {
+  const { chainId, account } = useActiveWeb3React()
+  const open = useModalOpen(ApplicationModal.FAUCET_POPUP)
+  const toggle = useToggleModal(ApplicationModal.FAUCET_POPUP)
+  const theme = useContext(ThemeContext)
+  const [rewardData, setRewardData] = useState<{ amount: BigNumber; tokenAddress: string; program: number }>()
+  const addPopup = useAddPopup()
+  const toggleWalletModal = useWalletModalToggle()
+  const { mixpanelHandler } = useMixpanel()
+  const allTokens = useAllTokens()
+  const token = useMemo(() => {
+    if (!chainId || !account) return
+    if (rewardData) {
+      if (rewardData.tokenAddress === '0') return ETHER as Token
+      if (isAddress(rewardData.tokenAddress)) return filterTokens(Object.values(allTokens), rewardData.tokenAddress)[0]
+    }
+    return ETHER as Token
+  }, [rewardData, chainId, account, allTokens])
+  const tokenLogo = useMemo(() => {
+    if (!chainId || !token) return
+    if (token === ETHER) return logo[chainId]
+    return getTokenLogoURL(token.address, chainId)
+  }, [chainId, token])
+  const claimRewardCallBack = async () => {
+    if (!rewardData) return
+    try {
+      const rawResponse = await fetch('https://reward.dev.kyberengineering.io/api/v1/rewards/claim', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ wallet: account, program: rewardData.program }),
+      })
+      const content = await rawResponse.json()
+      if (content) {
+        addPopup({
+          simple: {
+            title: `Request to Faucet - Submitted`,
+            success: true,
+            summary: `You will receive ${
+              rewardData?.amount ? getFullDisplayBalance(rewardData?.amount, token?.decimals) : 0
+            } ${token?.symbol} soon!`,
+          },
+        })
+        setRewardData(rw => {
+          if (rw) {
+            rw.amount = BigNumber.from(0)
+          }
+          return rw
+        })
+      }
+    } catch (error) {
+      console.log(error)
+    }
+  }
+  useEffect(() => {
+    if (!chainId || !account) return
+    const getRewardAmount = async () => {
+      try {
+        const { data } = await fetch(
+          `https://reward.dev.kyberengineering.io/api/v1/faucets?wallet=${account}&chainId=${chainId}`,
```

Comment:
do not hard code API URL like this, please move it to .env

Reply:
just did this when waiting for last api update